Boost 1.67 breaks the homebrew build

Boost 1.67 breaks the homebrew build.  Boost 1.67 came out at the same time as the 1.0.0 release,  looks like there will need to be a couple fixes to restore compatibility.    We will have to make sure the fixes work for both the new boost and previous versions.

Hi,
I just tried to install HELICS in Windows and Mac. I tried several ways to install it. But I always end up with this error.

> [ 13%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/tcp/TcpComms.cpp.o
/Users/belle/Documents/GitHub/HELICS-src/src/helics/core/tcp/TcpComms.cpp:140:15: error: 
      invalid operands to binary expression ('const boost::system::error_code'
      and 'nullptr_t')
    if (error != nullptr)
        ~~~~~ ^  ~~~~~~~
1 error generated.
make[2]: *** [src/helics/core/CMakeFiles/helics_core.dir/tcp/TcpComms.cpp.o] Error 1
make[1]: *** [src/helics/core/CMakeFiles/helics_core.dir/all] Error 2
make: *** [all] Error 2

I think new version of Boost might be the reason. Therefore, I posted it in here.
Thank you.

--- a/src/helics/core/tcp/TcpComms.cpp
+++ b/src/helics/core/tcp/TcpComms.cpp
@@ -137,7 +137,7 @@ void TcpComms::txPriorityReceive (std::shared_ptr<TcpConnection> /*connection*/,
                                   size_t bytes_received,
                                   const boost::system::error_code &error)
 {
-    if (error != nullptr)
+    if (error)
     {
         return;
     }

--- a/src/helics/core/tcp/TcpHelperClasses.cpp
+++ b/src/helics/core/tcp/TcpHelperClasses.cpp
@@ -110,7 +110,7 @@ void TcpRxConnection::close ()
     disconnected = true;
     boost::system::error_code ec;
     socket_.shutdown (boost::asio::ip::tcp::socket::shutdown_send, ec);
-    if (ec != nullptr)
+    if (ec)
     {
         std::cerr << "error occurred sending shutdown" << std::endl;
     }
@@ -196,7 +196,7 @@ void TcpConnection::close ()
     cancel ();
     boost::system::error_code ec;
     socket_.shutdown (boost::asio::ip::tcp::socket::shutdown_send, ec);
-    if (ec != nullptr)
+    if (ec)
     {
         // I don't know what to do with this, in practice this message is mostly spurious
         // but is seems I should do something with it, I just don't know what

